dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/frameworks/lksg/custom/LksgProcurementCategoriesComponent.kt

```diff
@@ -0,0 +1,86 @@
+import org.dataland.frameworktoolbox.specific.datamodel.elements.DataClassBuilder
+import org.dataland.frameworktoolbox.specific.fixturegenerator.elements.FixtureSectionBuilder
+import org.dataland.frameworktoolbox.specific.uploadconfig.elements.UploadCategoryBuilder
+import org.dataland.frameworktoolbox.specific.viewconfig.elements.SectionConfigBuilder
+import org.dataland.frameworktoolbox.specific.viewconfig.elements.getTypescriptFieldAccessor
+import org.dataland.frameworktoolbox.specific.viewconfig.functional.FrameworkDisplayValueLambda
+import org.dataland.frameworktoolbox.utils.typescript.TypeScriptImport
+
+/**
+ * Represents the ProcurementCategories component
+ */
+class LksgProcurementCategoriesComponent(
+    identifier: String,
+    parent: FieldNodeParent,
+) : ComponentBase(identifier, parent) {
+    override fun generateDefaultDataModel(dataClassBuilder: DataClassBuilder) {
+        requireDocumentSupportIn(setOf(NoDocumentSupport))
+        dataClassBuilder.addProperty(
+            identifier,
+            TypeReference(
+                "Map", isNullable,
+                listOf(
+                    TypeReference(
+                        "org.dataland.datalandbackend.frameworks.lksg.custom.ProcurementCategoryType",
+                        false,
+                    ),
+                    TypeReference(
+                        "org.dataland.datalandbackend.frameworks.lksg.custom.LksgProcurementCategory",
+                        false,
+                    ),
+                ),
+            ),
+        )
+    }
+
+    override fun generateDefaultViewConfig(sectionConfigBuilder: SectionConfigBuilder) {
+        requireDocumentSupportIn(setOf(NoDocumentSupport))
+        sectionConfigBuilder.addStandardCellWithValueGetterFactory(
+            this,
+            FrameworkDisplayValueLambda(
+                "formatLksgProcurementCategoriesForDisplay(${getTypescriptFieldAccessor(true)}, \"${
+                    StringEscapeUtils.escapeEcmaScript(
+                        label,
+                    )
+                }\")",
+                setOf(
+                    TypeScriptImport(
+                        "formatLksgProcurementCategoriesForDisplay",
+                        "@/components/resources/dataTable/conversion/lksg/LksgValueGetterFactories",
+                    ),
+                ),
+            ),
+        )
+    }
+
+    override fun generateDefaultUploadConfig(uploadCategoryBuilder: UploadCategoryBuilder) {
+        requireDocumentSupportIn(setOf(NoDocumentSupport))
+        uploadCategoryBuilder.addStandardUploadConfigCell(
+            component = this,
+            uploadComponentName = "ProcurementCategoriesFormField",
+        )
+    }
+
+    override fun generateDefaultFixtureGenerator(sectionBuilder: FixtureSectionBuilder) {
+        requireDocumentSupportIn(setOf(NoDocumentSupport))
+        val fixtureExperssion = if (isNullable) {
```

Comment:
This has nothing to do with nullability of the field.

Reply:
You are only selecting a different amount of procurementcategories throught this.

Reply:
wenn isNullable true ist, dann können 0-3 categories generiert werden.
wenn es false ist, dann sind es garantiert 3 cateogires. das erschien mir einfacher als explizit 1-3 categories zu implementieren.

Reply:
Ich dachte im fall isNullable true soll sowas wie `this.valueOrNull(dataGenerator.generateProcurementCategories())` verwendet werden, weil es ja um die nullability von dem Feld selbst geht und nicht um den Status des Objekts.

Reply:
Ist wie besprochen geändert